mousestyles/ultradian/__init__.py

```diff
@@ -0,0 +1,456 @@
+    ts: pandas.tseries
+        a pandas time series of length (#day)*24(hour)*60(minute)/n
+    """
+    # Input Check
+    if (not isinstance(strain, int)) or (strain < 0):
+        raise ValueError(
+            'Strain must be a non-negative integer')
+    if (not isinstance(mouse, int)) or (mouse < 0):
+        raise ValueError(
+            'Mouse value must be a non-negative integer')
+    if (not isinstance(bin_width, int)) or bin_width < 0 or bin_width > 1440:
+        raise ValueError(
+            'Bin width (minutes) must be a non-negative integer below 1440')
+
+    # determine number of days
+    intervals = data.load_intervals('IS')
+    mouse_data = intervals.loc[
+        (intervals['strain'] == strain) & (intervals['mouse'] == mouse)]
+    days = sorted(np.unique(mouse_data['day']))
+
+    # build data frame
+    bin_count = int(24 * 60 / bin_width)
+    time_movements = np.repeat(0.0, bin_count * len(days))
+    bin_length = bin_width * 60
+    for j in days:
+        M = data.load_movement(strain, mouse, day=int(j))
+        distance_df = pd.DataFrame({"start": M["t"].values[0:-1],
+                                    "end": M["t"].values[1:],
+                                    "distance":
+                                    np.linalg.norm(M[["x", "y"]].values[1:] -
+                                                   M[["x", "y"]].values[0:-1],
+                                                   axis=1)})
+        start_end = data.load_start_time_end_time(strain, mouse, j)
+        start = np.asarray(distance_df['start']) - start_end[0]
+        end = np.asarray(distance_df['end']) - start_end[0]
+        dist = distance_df['distance']
+        for i in range(len(start)):
+            start_time = start[i]
+            end_time = end[i]
+            start_index = int(start_time / (bin_width * 60))
+            end_index = int(end_time / (bin_width * 60))
+            if start_index == end_index:
+                time_movements[start_index + j *
+                               bin_count] += dist[i]
+            else:
+                time_movements[
+                    end_index + j * bin_count] += end_time % \
+                    bin_length / (end_time - start_time) * dist[i]
+                time_movements[
+                    start_index + j * bin_count] += dist[i] - \
+                    end_time % bin_length / (end_time - start_time) * dist[i]
+    ts = pd.Series(time_movements, index=pd.date_range(
+        '01/01/2014', periods=len(time_movements),
+        freq=str(bin_width) + 'min'))
+    return(ts)
+
+
+def aggregate_data(feature, bin_width):
+    r"""
+    Aggregate all the strains and mouses with any feature together
+    in one dataframe. It combines the results you got from
+    aggregate_movements and aggregate_interval. It will return
+    a dataframe with three variables: mouse, strain, feature and hour.
+
+    Parameters
+    ----------
+    feature :
+        {"AS", "F", "IS", "M_AS", "M_IS", "W"}
```

Comment:
@HaoLyuKrystal  The feature now will not include 'IS' ? the only thing we have together with other functions is "AS", "F", "M_AS", "M_IS", "W" and the 'Distance' I think =0